### PR TITLE
[NFC] add utils for PSV.

### PR DESCRIPTION
**Pull request by python3kgae**
Add init functions to help build PSVSignatureElement0/PSVRuntimeInfo/PSVResourceInfo.

Add print function for DxilPipelineStateValidation.

Add new option -dumppsv to dxa for testing.

For #6817.

### CODE REVIEW

File: include/dxc/DxilContainer/DxilPipelineStateValidation.h [hunk 1]
```diff
@@ -16,6 +16,12 @@
 #include <cstring>
 #include <stdint.h>
 
+#include <assert.h>
```
Comment (damyanp) [inline include/dxc/DxilContainer/DxilPipelineStateValidation.h:19]: nit: why the line break between this and the other includes?

File: include/dxc/DxilContainer/DxilPipelineStateValidation.h [hunk 2]
```diff
@@ -266,7 +274,8 @@ struct PSVComponentMask {
     if (ComponentIndex < NumVectors * 4)
       Mask[ComponentIndex >> 5] &= ~(1 << (ComponentIndex & 0x1F));
   }
-  bool IsValid() { return Mask != nullptr; }
+  bool IsValid() const { return Mask != nullptr; }
+  void print(llvm::raw_ostream &, const char *, const char *) const;
```
Comment (damyanp) [inline include/dxc/DxilContainer/DxilPipelineStateValidation.h:278]: Looks like there are two different naming conventions being used here.  It'd be good to try and figure out what one if most prevalent in this file and follow that one?  Based on a quick look, it seems like that would make this `Print`.

This applies to all the new methods in this PR, I won't call them all out, although if we agree we should follow this convention and you let me know you think you've updated them all I could do that if you like.

**Review by damyanp — APPROVED** [review]
It looks like quite a lot of code has moved around...I couldn't quite figure out if that's actually required to add the printing functionality, or it is was just a cleanup.  I haven't done a careful diff of the moved code, so I'm going to assume that there truly is NFC here.  I'm guessing that without the PSV dumping functionality added here that there's minimal existing test coverage of this.

File: include/dxc/DxilContainer/DxilPipelineStateValidation.h (hunk 2 above)
Comment (python3kgae) [reply include/dxc/DxilContainer/DxilPipelineStateValidation.h]: Updated.

File: include/dxc/DxilContainer/DxilPipelineStateValidation.h (hunk 1 above)
Comment (python3kgae) [reply include/dxc/DxilContainer/DxilPipelineStateValidation.h]: Fixed.

**Review by bob80905 — APPROVED** [review]
Nit: Several of the files don't end with a newline

**Pull request merged**